core/src/main/java/io/questdb/cutlass/auth/Authenticator.java

```diff
@@ -49,7 +51,7 @@ default void close() {
 
     int handleIO() throws AuthenticatorException;
 
-    void init(int fd, long recvBuffer, long recvBufferLimit, long sendBuffer, long sendBufferLimit);
+    void init(@Nullable Socket socket, long recvBuffer, long recvBufferLimit, long sendBuffer, long sendBufferLimit);
```

Comment:
how can socket be `null`?

Reply:
Some of the I/O contexts need to clean it up, e.g. `PGConnectionContext`. See https://github.com/questdb/questdb/blob/c9c85f114ab5a3bbedd9656031b5fc025893db61/core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java#L451

Reply:
Changed in 5ee92af47fde755e6e35cfd805020f8dc848e34f